rm-datasource/src/main/java/io/seata/rm/datasource/AsyncWorker.java

```diff
@@ -171,12 +175,28 @@ private void doBranchCommits() {
                     continue;
                 }
                 List<Phase2Context> contextsGroupedByResourceId = entry.getValue();
+                Set<String> xids = new LinkedHashSet<>(UNDOLOG_DELETE_LIMIT_SIZE);
+                Set<Long> branchIds = new LinkedHashSet<>(UNDOLOG_DELETE_LIMIT_SIZE);
                 for (Phase2Context commitContext : contextsGroupedByResourceId) {
+                    xids.add(commitContext.xid);
+                    branchIds.add(commitContext.branchId);
+                    int maxSize = xids.size() > branchIds.size() ? xids.size() : branchIds.size();
+                    if(maxSize % UNDOLOG_DELETE_LIMIT_SIZE == 0){
```

Comment:
`maxSize % UNDOLOG_DELETE_LIMIT_SIZE == 0` seems `maxSize==UNDOLOG_DELETE_LIMIT_SIZE` more clear? because you will clear the collection

Reply:
yes，already done !